gwcs/geometry.py

```diff
@@ -74,29 +81,79 @@ class SphericalToCartesian(Model):
     n_inputs = 2
     n_outputs = 3
 
-    def __init__(self, **kwargs):
+    def __init__(self, wrap_phi_at=360, theta_def='elevation', **kwargs):
+        """
+        Parameters
+        ----------
+        wrap_phi_at : {180, 360}, optional
+            Specifies the range of the azimuthal angle. When ``wrap_phi_at`` is
+            180, azimuthal angle will have a range of ``[-180, 180)`` and
+            when ``wrap_phi_at`` is 360 (default), the azimuthal angle will have
+            a range of ``[0, 360)``
+
+        theta_def : {'elevation', 'inclination', 'polar'}, optional
+            Specifies the definition used for the ``theta``: 'elevation' from
+            the reference plane or 'inclination' (or 'polar') angle measured
+            from the pole.
+
+        """
         super().__init__(**kwargs)
+        self.wrap_phi_at = wrap_phi_at
```

Comment:
This and the attribute below should be set to the private variables, e.g.  `self._wrap_phi_at = wrap_phi_at`.

Reply:
It is in the setter, isn't it?

Reply:
It needs to be here as well

Reply:
I'm not sure it does since the setter will define it and the setter is called in the __init__. Where will this be be a problem?

Reply:
@nden @perrygreenfield 
It looks like it is a pretty common thing to do to avoid duplication of code. Some other people seem to have had concerns due to `pylint` warnings, e.g., https://stackoverflow.com/q/32383581

I think the concern that @nden has is that when one looks at the code in `__init__()`, they will see `self.x = x` (to make it shorter), and they will think that (or does @nden think that  it actually is?) `self.x` is an _attribute_ instead of _property_. I think in Python there are many places where behavior is re-defined and, in my opinion, this should be OK. 

One possible improvement to the clarity of the code would be to do something that is done here: https://github.com/spacetelescope/jwst/blob/c6303abcfb8b200a9fca9d69378c857a8ed0ed25/jwst/regtest/regtestdata.py#L34